src/query/functions/src/scalars/map.rs

```diff
@@ -244,4 +258,134 @@ pub fn register(registry: &mut FunctionRegistry) {
                     .any(|(k, _)| k == key)
             },
         );
+
+    registry.register_function_factory("map_pick", |_, args_type: &[DataType]| {
+        if args_type.len() < 2 {
+            return None;
+        }
+
+        if !matches!(args_type[0], DataType::Map(_) | DataType::EmptyMap) {
+            return None;
+        }
+
+        let inner_key_type = match args_type.first() {
+            Some(DataType::Map(m)) => m.as_tuple().map(|tuple| &tuple[0]),
+            _ => None,
+        };
+        let key_match = match args_type.len() {
+            2 => args_type.get(1).map_or(false, |t| match t {
```

Comment:
if the arguments number  is two, the second argument may be an array type or a key type.